### PR TITLE
Fix correctness checks on Travis and commit hooks

### LINKED ISSUE #94: We should warn about unknown `[[Tags]]` on tests
Currently they are ignored and all the used ones are optional, so there's no protection against typos or missed updates from the old code.

### LINKED ISSUE #178: Trigger all regular expression tests failures at once
Currently the test will abort at the first failure

### LINKED ISSUE #179: Have Travis check for 'Ls'
Ls should be List

### PR DESCRIPTION
**Pull request by nilsdeppe**
Fix #94 
Fix #178 
Fix #179

### CODE REVIEW

File: src/DataStructures/Tensor/EagerMath/Determinant.hpp [hunk 1]
```diff
@@ -126,7 +126,8 @@ struct DeterminantImpl<Symmetry<1, 1>, Index, Requires<Index::dim == 4>> {
 };
 }  // namespace detail
 
-/*! \ingroup Functors
```
Comment (nilsdeppe) [inline src/DataStructures/Tensor/EagerMath/Determinant.hpp:129]: also search for `defgroup`

File: src/DataStructures/Tensor/EagerMath/Determinant.hpp [hunk 2]
```diff
@@ -126,7 +126,8 @@ struct DeterminantImpl<Symmetry<1, 1>, Index, Requires<Index::dim == 4>> {
...
+/*!
+ * \ingroup Functions
```
Comment (nilsdeppe) [inline src/DataStructures/Tensor/EagerMath/Determinant.hpp]: Remove `Functions` and `Functions for GR` groups. Move this to `Tensor`

File: tools/Hooks/pre-commit.sh [hunk 3]
```diff
@@ -205,6 +205,19 @@ if [[ $found_incorrect_list_name != "" ]]; then
 fi
 
 ###############################################################################
+# Check for Ls because of a preference not to use it as short form for List
```
Comment (kidder) [inline tools/Hooks/pre-commit.sh]: RIP

**Review by kidder — CHANGES_REQUESTED** [review]

File: tools/Hooks/pre-commit.sh (hunk 3 above)
Comment (nilsdeppe) [reply tools/Hooks/pre-commit.sh]: PIR

File: tools/CheckFiles.sh [hunk 4]
```diff
@@ -107,38 +112,82 @@ find ./ -type f                                                              \
...
            "SPECTRE_ROOT/tests/Unit/TestHelpers.hpp instead:\n"
     echo "$found_bad_approx"
+    found_error=1
+fi
+
+###############################################################################
+# Check for Doxygen group inclusions on the starting line
+found_incorrect_ingroup_syntax=`
+find ./ -type f -name '*.[ch]pp' -not -path './build*' \
+    | xargs grep --with-filename -n '/\*\![ ]*\\\\ingroup'`
```
Comment (wthrowe) [inline tools/CheckFiles.sh]: Would it make sense to just ban any "/*! \", rather than checking for particular commands?

Comment (wthrowe) [reply tools/CheckFiles.sh]: That was supposed to have a backslash in it, but it seems to have been eaten by GitHub.

Comment (nilsdeppe) [reply tools/CheckFiles.sh]: I'd be happy with banning "/*!.+" (if grep can do a line by line search and does not match newlines with `.`) The `/*!` comments are useful (needed?) for multiline math and I have developed a personal preference for them.

Comment (wthrowe) [reply tools/CheckFiles.sh]: I intended to suggest banning slash-star-bang-space*-backslash, but GitHub stripped the backslash (and I forgot the *, or + if you prefer).  Plain slash-star-bang should certainly be allowed.

Comment (nilsdeppe) [reply tools/CheckFiles.sh]: Okay, we are in agreement then, I will make the change.

File: src/DataStructures/Tensor/EagerMath/Determinant.hpp [hunk 5]
```diff
@@ -126,7 +126,8 @@ struct DeterminantImpl<Symmetry<1, 1>, Index, Requires<Index::dim == 4>> {
 };
 }  // namespace detail
 
-/*! \ingroup Functors
+/*!
+ * \ingroup Tensor
```
Comment (fmahebert) [inline src/DataStructures/Tensor/EagerMath/Determinant.hpp:130]: Consistency question... why put this in group Tensor when you suggested `determinant_and_inverse` be in group NumericalAlgorithms? Seems to me that `determinant` and `determinant_and_inverse` belong together.

Comment (kidder) [reply src/DataStructures/Tensor/EagerMath/Determinant.hpp:130]: I agree they should be consistent, but in group Tensor

Comment (fmahebert) [reply src/DataStructures/Tensor/EagerMath/Determinant.hpp:130]: OK by me. If @nilsdeppe agrees I will make the corresponding change in #137

Comment (nilsdeppe) [reply src/DataStructures/Tensor/EagerMath/Determinant.hpp:130]: I agree! I agree!

File: tools/CheckFiles.sh [hunk 6]
```diff
@@ -107,38 +112,69 @@ find ./ -type f                                                              \
...
     echo "SPECTRE_ROOT/tools/CheckFiles.sh"
     printf "Found occurrences of Approx, must use approx from " \
            "SPECTRE_ROOT/tests/Unit/TestHelpers.hpp instead:\n"
     echo "$found_bad_approx"
+    found_error=1
+fi
+
+###############################################################################
+# Check for Doxygen group inclusions on the starting line
+found_incorrect_ingroup_syntax=`
```
Comment (kidder) [inline tools/CheckFiles.sh]: rename variable to found_bad_doxygen_syntax and update comment since this will error on anything on the starting line

**Review by kidder — CHANGES_REQUESTED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**